### PR TITLE
limit order imporovement

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**

### CODE REVIEW

File: src/components/swapv2/LimitOrder/Tutorial.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            text={
+              <Trans>
+                Place orders on thousands on tokens for{' '}
+                <Text as="span" fontWeight={'500'} color={theme.primary}>
+                  <Trans>FREE</Trans>
+                </Text>
+              </Trans>
+            }
+          />
+          <ContentItem text={t`Get your orders filled without paying any gas fees`} />
```
Comment (namgold) [inline src/components/swapv2/LimitOrder/Tutorial.tsx:78]:
```suggestion
          <ContentItem text={t`Get your orders filled without paying any gas fees.`} />
```
:dungz:

File: src/components/swapv2/LimitOrder/useValidateInputError.tsx [hunk 2]
```diff
@@ -39,10 +44,23 @@ const useValidateInputError = ({
...
+          fractionDigits: 6,
+          allowNegative: true,
+        })
+        return (
+          <Text sx={{ cursor: 'pointer' }}>
+            <Trans>
+              You don&apos;t have sufficient {currencyIn?.symbol} balance. After your active orders, you have{' '}
+              <Text as="b" color={theme.primary} onClick={() => setInputValue(remainBalance.toExact())}>
+                {Number(formatNum) !== 0 ? '~' : ''}
+                {formatNum} {currencyIn?.symbol}
```
Comment (namgold) [inline src/components/swapv2/LimitOrder/useValidateInputError.tsx:58]: Posible displayed as `0 undefined`?

File: src/components/swapv2/LimitOrder/useValidateInputError.tsx [hunk 3]
```diff
@@ -39,10 +44,23 @@ const useValidateInputError = ({
...
       const remainBalance = parsedActiveOrderMakingAmount ? balance?.subtract(parsedActiveOrderMakingAmount) : undefined
       if (parseInputAmount && remainBalance?.lessThan(parseInputAmount)) {
-        const formatNum = formatNumberWithPrecisionRange(parseFloat(remainBalance.toFixed(3)), 0, 10)
-        return t`You don't have sufficient ${currencyIn?.symbol} balance. After your active orders, you have ${
-          Number(formatNum) !== 0 ? '~' : ''
-        }${formatNum} ${currencyIn?.symbol} left.`
+        const formatNum = formatDisplayNumber(remainBalance, {
+          style: 'currency',
+          fractionDigits: 6,
+          allowNegative: true,
```
Comment (namgold) [inline src/components/swapv2/LimitOrder/useValidateInputError.tsx:50]:
```suggestion
          allowNegative: true,
          allowZero: false,
```

Comment (namgold) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:50]: why would this place need allow negative?

File: src/pages/CrossChain/SwapForm/index.tsx [hunk 4]
```diff
@@ -189,7 +189,9 @@ export default function SwapForm() {
       onTracking(MIXPANEL_TYPE.CROSS_CHAIN_TXS_SUBMITTED)
       setInputAmount('')
       setSwapState(state => ({ ...state, attemptingTxn: false, txHash: tx.hash }))
-      const tokenAmountOut = uint256ToFraction(outputAmount, currencyOut.decimals).toSignificant(6)
+      const tokenAmountOut = uint256ToFraction(outputAmount, currencyOut.decimals).toSignificant(
+        Math.min(6, currencyOut.decimals),
+      )
```
Comment (namgold) [inline src/pages/CrossChain/SwapForm/index.tsx:194]: consider use `getFullDisplayBalance` instead

File: src/components/swapv2/LimitOrder/useValidateInputError.tsx (hunk 3 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:50]: can be negative. 
can be zero

File: src/components/swapv2/LimitOrder/useValidateInputError.tsx (hunk 2 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:58]: undefined will not display

File: src/components/swapv2/LimitOrder/Tutorial.tsx (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/Tutorial.tsx:78]: 🚀

File: src/components/swapv2/LimitOrder/useValidateInputError.tsx (hunk 3 above)
Comment (namgold) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:50]: This flag is for allowing display zero or negative or fallback with `--`. I saw this formatNum is comparing with zero and display it as ~ below so this num is not allow to show zero?

Comment (namgold) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:50]: Might be renaming to `allowDisplayNegative` & `allowDisplayZero` will be better to read

Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:51]: I dont want to use allowZero for this case
Just want to add prefix `~`
`~` amount if (amount != 0) else amount

Comment (namgold) [reply src/components/swapv2/LimitOrder/useValidateInputError.tsx:51]: got it

File: src/pages/CrossChain/SwapForm/index.tsx [hunk 5]
```diff
@@ -189,7 +190,7 @@ export default function SwapForm() {
       onTracking(MIXPANEL_TYPE.CROSS_CHAIN_TXS_SUBMITTED)
       setInputAmount('')
       setSwapState(state => ({ ...state, attemptingTxn: false, txHash: tx.hash }))
-      const tokenAmountOut = uint256ToFraction(outputAmount, currencyOut.decimals).toSignificant(6)
+      const tokenAmountOut = getFullDisplayBalance(outputAmount, currencyOut.decimals, 6)
```
Comment (namgold) [inline src/pages/CrossChain/SwapForm/index.tsx:193]: tested?

**Review by namgold — APPROVED** [review]

File: src/pages/CrossChain/SwapForm/index.tsx (hunk 5 above)
Comment (nguyenhoaidanh) [reply src/pages/CrossChain/SwapForm/index.tsx:193]: yes sir

**Pull request merged**